compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h

```diff
@@ -32,42 +33,35 @@ void addPadBasedPassPipeline(OpPassManager &passManager,
 
 /// Create a pass to do some rewrites that help bridging the path to AIR/AIE
 /// lowering.
-std::unique_ptr<OperationPass<>> createAMDAIEBridgeToAIRPass();
+std::unique_ptr<Pass> createAMDAIEBridgeToAIRPass();
 
 /// Pass to bufferizes the targeted operation and materializes the result in a
 /// new allocation.
-/// Currently using paddingLevel = 0 to indicate the target operation is
-/// packing.
-/// TODO: Find a better way to do the pass without using paddingLevel.
-std::unique_ptr<OperationPass<func::FuncOp>>
-createAMDAIEBufferizeToAllocationPass(int64_t memorySpace = 1,
-                                      int64_t paddingLevel = 0);
+std::unique_ptr<Pass> createAMDAIEBufferizeToAllocationPass(
+    const AMDAIEBufferizeToAllocationOptions &options = {});
 
 /// Create pass to invoke several cleanup and canonicalization patterns.
-std::unique_ptr<OperationPass<func::FuncOp>> createAMDAIECleanupPass();
+std::unique_ptr<Pass> createAMDAIECleanupPass();
 
 /// Create pass calling the dynamic pipeline for AMDAIE.
-std::unique_ptr<OperationPass<IREE::HAL::ExecutableVariantOp>>
-createAMDAIELowerExecutableTargetPass();
+std::unique_ptr<Pass> createAMDAIELowerExecutableTargetPass();
 
 /// Create a pass to lower workgroup count region of entry point operations.
-std::unique_ptr<OperationPass<IREE::HAL::ExecutableVariantOp>>
-createAMDAIELowerWorkgroupCountPass();
+std::unique_ptr<Pass> createAMDAIELowerWorkgroupCountPass();
 
 /// Create a pass to pack and transpose the linalg op.
-std::unique_ptr<OperationPass<func::FuncOp>> createAMDAIEPackAndTransposePass(
-    int64_t packLevel = 1);
+std::unique_ptr<Pass> createAMDAIEPackAndTransposePass(
+    const AMDAIEPackAndTransposeOptions &options = {});
```

Comment:
Nit: I am not sure what happens if you initialize a reference this way. It maybe fine, but I have seen this segfault sometimes. Drop the `const &` here and everywhere.